src/features/QueueProducer.js

```diff
@@ -0,0 +1,192 @@
+      // create session
+      try {
+          producer.session = solace.SolclientFactory.createSession({
+              // solace.SessionProperties
+              url:      hosturl,
+              vpnName:  vpn,
+              userName: username,
+              password: pass,
+              publisherProperties: {
+                acknowledgeMode: solace.MessagePublisherAcknowledgeMode.PER_MESSAGE,
+              }          
+          });
+      } catch (error) {
+          producer.log(error.toString());
+      }
+      // define session event listeners
+      producer.session.on(solace.SessionEventCode.UP_NOTICE, function (sessionEvent) {
+          producer.log('=== Successfully connected and ready to send messages. ===');
+          producer.sendMessage();
+      });
+      producer.session.on(solace.SessionEventCode.CONNECT_FAILED_ERROR, function (sessionEvent) {
+          producer.log('Connection failed to the message router: ' + sessionEvent.infoStr +
+              ' - check correct parameter values and connectivity!');
+      });
+      producer.session.on(solace.SessionEventCode.DISCONNECTED, function (sessionEvent) {
+          producer.log('Disconnected.');
+          if (producer.session !== null) {
+              producer.session.dispose();
+              producer.session = null;
+          }
+      });
+      producer.session.on(solace.SessionEventCode.ACKNOWLEDGED_MESSAGE, function (sessionEvent) {
+          producer.log('Delivery of message with correlation key = ' +
+              sessionEvent.correlationKey.id + ' confirmed.');
+          producer.exit();
+      });
+      producer.session.on(solace.SessionEventCode.REJECTED_MESSAGE_ERROR, function (sessionEvent) {
+          producer.log('Delivery of message with correlation key = ' +
+              sessionEvent.correlationKey.id + ' rejected, info: ' + sessionEvent.infoStr);
+          producer.exit();
+      });
+
+      // connect the session
+      try {
+          producer.session.connect();
+      } catch (error) {
+          producer.log(error.toString());
+      }
+  };
+
+  // Sends one message
+  producer.sendMessage = function () {
+      if (producer.session !== null) {
+          var messageText = 'Sample Message';
+          var message = solace.SolclientFactory.createMessage();
+          producer.log('Sending message "' + messageText + '" to queue "' + producer.queueName + '"...');
+          message.setDestination(solace.SolclientFactory.createDurableQueueDestination(producer.queueName));
+          message.setBinaryAttachment(messageText);
+          message.setDeliveryMode(solace.MessageDeliveryModeType.PERSISTENT);
+          // Define a correlation key object
+          const correlationKey = {
+              name: "MESSAGE_CORRELATIONKEY",
+              id: Date.now()
+          };
+          message.setCorrelationKey(correlationKey);
+
+          try {
+              // Delivery not yet confirmed. See ConfirmedPublish.js
```

Comment:
same comment here - isint delivery confirmed in the `solace.SessionEventCode.ACKNOWLEDGED_MESSAGE`

Reply:
Correct. The CorrelationKey is optional - just updated the comment in this regard. 

          // OPTIONAL: You can set a correlation key on the message and check for the correlation
          // in the ACKNOWLEDGE_MESSAGE callback.
          // Define a correlation key object

If you think the correlation key discussion is not required, we can remove it.